pkg/api/proxyConfig.go

```diff
@@ -0,0 +1,143 @@
+package api
+
+import (
+	"context"
+	"github.com/google/uuid"
+	"github.com/permitio/permit-golang/pkg/config"
+	"github.com/permitio/permit-golang/pkg/errors"
+	"github.com/permitio/permit-golang/pkg/models"
+	"github.com/permitio/permit-golang/pkg/openapi"
+	"go.uber.org/zap"
+)
+
+type ProxyConfigs struct {
+	permitBaseApi
+}
+
+func NewProxyConfigsApi(client *openapi.APIClient, config *config.PermitConfig) *ProxyConfigs {
+	return &ProxyConfigs{
+		permitBaseApi{
+			client: client,
+			config: config,
+			logger: config.Logger,
+		},
+	}
+}
+
+// List all ProxyConfigs in the organization, requires Project level API key, or higher.
+// Usage Example:
+// `ProxyConfigs, err := PermitClient.Api.ProxyConfigs.List(ctx, 1, 10)`
+func (p *ProxyConfigs) List(ctx context.Context, page int, perPage int) ([]models.ProxyConfigRead, error) {
+	perPageLimit := int32(DefaultPerPageLimit)
+	if !isPaginationInLimit(int32(page), int32(perPage), perPageLimit) {
+		err := errors.NewPermitPaginationError()
+		p.logger.Error("error listing ProxyConfigs - max per page: "+string(perPageLimit), zap.Error(err))
+		return nil, err
+	}
+	err := p.lazyLoadPermitContext(ctx, config.EnvironmentAPIKeyLevel)
```

Comment:
it is stated that it required project level api key in line 27 but here it requires environment level api key